jboss/container/eap/prometheus/config/7.1/artifacts/opt/jboss/container/prometheus/etc/jmx-exporter-config.yaml

```diff
@@ -0,0 +1,363 @@
+  name: datasources_$3_total
+  help: Datasource JDBC Metrics $3
+  value: $4
+  valueFactor: 1
+  labels:
+    type: $1
+    name: $2
+  attrNameSnakeCase: true
+  type: COUNTER
+- pattern: "jboss.as<subsystem=datasources, *(data-source|xa-data-source)=(.+), *statistics=pool><>(active_count|available_count|in_use_count|max_used_count): (.+)"
+  name: datasources_$3
+  help: Datasource Pool Metrics $3
+  value: $4
+  valueFactor: 1
+  labels:
+    type: $1
+    name: $2
+  attrNameSnakeCase: true
+  type: GAUGE
+- pattern: "jboss.as<subsystem=datasources, *(data-source|xa-data-source)=(.+), *statistics=pool><>(average_blocking_time|average_creation_time|average_get_time|max_creation_time|max_get_time|max_wait_time): (.+)"
+  name: datasources_$3_seconds
+  help: Datasource Pool Metrics $3
+  value: $4
+  valueFactor: .001
+  labels:
+    type: $1
+    name: $2
+  attrNameSnakeCase: true
+  type: GAUGE
+- pattern: "jboss.as<subsystem=datasources, *(data-source|xa-data-source)=(.+), *statistics=pool><>(blocking_failure_count|created_count|destroyed_count|idle_count|timed_out|wait_count): (.+)"
+  name: datasources_$3_total
+  help: Datasource Pool Metrics $3
+  value: $4
+  valueFactor: 1
+  labels:
+    type: $1
+    name: $2
+  attrNameSnakeCase: true
+  type: COUNTER
+- pattern: "jboss.as<subsystem=datasources, *(data-source|xa-data-source)=(.+), *statistics=pool><>(total_blocking_time|total_creation_time|total_get_time): (.+)"
+  name: datasources_$3_seconds_total
+  help: Datasource Pool Metrics $3
+  value: $4
+  valueFactor: .001
+  labels:
+    type: $1
+    name: $2
+  attrNameSnakeCase: true
+  type: COUNTER
+
+# Transactions
+- pattern: "jboss.as<subsystem=transactions><>(number_of_inflight_transactions): (.+)"
+  name: transactions_$1
+  help: Transactions Metrics $1
+  value: $2
+  valueFactor: 1
+  attrNameSnakeCase: true
+  type: GAUGE
+- pattern: "jboss.as<subsystem=transactions><>(number_of_aborted_transactions|number_of_application_rollbacks|number_of_committed_transactions|number_of_heuristics|number_of_nested_transactions|number_of_resource_rollbacks|number_of_timed_out_transactions|number_of_transactions): (.+)"
+  name: transactions_$1_total
+  help: Transactions Metrics $1
+  value: $2
+  valueFactor: 1
+  attrNameSnakeCase: true
+  type: COUNTER
+
+# JMS
+# Queues
+- pattern: "jboss.as<subsystem=messaging-activemq, *server=(.+), *jms-queue=(.+)><>(consumer_count|delivering_count|message_count|scheduled_count): (.+)"
+  name: messaging_activemq_jms_queue_$3
+  help: Messaging JMS Queue Metrics $3
+  value: $4
+  valueFactor: 1
+  labels:
+    server: $1
+    name: $2
+  attrNameSnakeCase: true
+  type: GAUGE
+- pattern: "jboss.as<subsystem=messaging-activemq, *server=(.+), *jms-queue=(.+)><>(messages_added): (.+)"
+  name: messaging_activemq_jms_queue_$3_total
```

Comment:
for consistency, the name should be `messaging_activemq_$3_total`

and the labels should have:

```
   type: jms-queue
```

(same remark goes for jms-topic resources)

Reply:
I wasn't sure about that, since the metric set differs between queues and topics.  I can update the PR.